Fix EZP-24553: no results for exact phrases with camelCase words

joaoinacio opened the pull request:
JIRA: https://jira.ez.no/browse/EZP-24553

When performing an exact match for some phrases, no results are returned.

#### Problem: ####
After investigation it seems the cause of the problem is caused by the existence of "camelCase" words in the sentence (eg: `eZ`)

The wordDelimiter filter in the `index` analyzer performs `splitOnCaseChange` and `concatenateWords` ( which causes "eZ Publish Platform" to be indexed as "e / ez / z | publish | platform" );
However, the `query` analyzer does not perform these operations, so the terms end up being different and not matching.

#### Solution: ####
Changing either the index or query analyzers to the same camelCase splitting behavior.

ping @paulborgermans

peterkeung commented:
If the alternative means that you modify the index analyzer to NOT split on camel case, then I vote for the alternative.  There are definitely times where you want to find something like "BlackBerry" and not have it match "Black Berry".  It is worth having to re-index to avoid bad matches when you were explicitly quoting the string.

bgamrat commented:
I agree with Peter - it is better to reindex and avoid bad matches.

andrerom commented:
I think I agree, but would like some input from @paulborgermans, he should be back next week afaik.

joaoinacio commented:
Thanks for the feedback; indeed, changing the index instead makes perfect sense, so should most likely be updated accordingly.
As noted, waiting for some additional input here from pb in case something was overlooked.
In any case everything looked good on my tests, besides the very minor BC (is it even one?)

paulborgermans commented:
+1, I am ok with the pull request, it should indeed be consistent. 

But also be aware that preferences on the analysis steps are up to the particular implementation. So camel case splitting may be turned of for certain customer projects, but active for others.

There is no big "truth" on what analysis should do out of the box/per default

joaoinacio commented:
Updated.
@paulborgermans: Thanks for the feedback, indeed the configuration may be a matter of particular needs and modified accordingly; this modifies the default behavior to be more "natural" with most common use cases.
@andrerom: Would there be any need to for doc. here?

bdunogier commented:
+1
